### PR TITLE
Overhaul E2E tests token activation logic

### PR DESCRIPTION
**Pull request by nemanjaglumac**
## The problem
Using `MB_PREMIUM_EMBEDDING_TOKEN` means that the token cannot be updated at runtime (say using `PUT` request). This makes our tests very rigid.

```
SETUP WITH `MB_PREMIUM_EMBEDDING_TOKEN` with all features active
.
├── OSS artifact (Token doesn't have any effect)
│   ├── Ambiguous tests run
│   ├── Tests tagged specifically with `@OSS` run
│   └── Tests inside `describeEE` are ignored
│
└──EE artifact
   ├── Tests tagged specifically with `@OSS` are ignored
   ├── All tests run with hard-coded PRO token
   └── It is not possible to unset or update the token at runtime

*To unset the token or to test token with different features, you need additional setup (or multiple setups).
```

## The solution
- Start Metabase instance needed for E2E tests without `MB_PREMIUM_EMBEDDING_TOKEN`
- Explicitly set the token (via `PUT` request) at runtime in tests that need specific features
- Provide ANY token that'll be needed at runtime as `CYPRESS_`
    - This trick makes such environment variable available to all tests at runtime
    - Example: You can get the value of `CYPRESS_FOO="bar"` later in tests with `Cypress.env("FOO")`
    - https://docs.cypress.io/guides/guides/environment-variables#Option-3-CYPRESS_

```
SETUP WITHOUT A TOKEN
.
├── OSS artifact
│   ├── Ambiguous tests run
│   ├── Tests tagged specifically with `@OSS` run
│   └── Tests inside `describeEE` are ignored
│
└──EE artifact
   ├── Tests tagged specifically with `@OSS` are ignored
   ├── Ambiguous tests run (without a token)
   └── Enterprise-specific tests (`describeEE`)
       ├── Run without a token
       └── Run with token set at runtime
           ├── No features (Starter)
           ├── All features (PRO/EE)
           ├── X/Y/Z features (arbitrary feature flag combo)
           └── Invalid / expired

* No need for multiple setups!
```

## Trade-offs
- Additional work/code needed to set the right token
- Makes conditional checks inside tests difficult (which is a good thing, imo; we should strive to have clean tests without too much if/else logic)
- We will need to work on some good helpers to introduce dev experience and to reduce fragility
- A bunch of tests needed to be updated
    - This also revealed some fragile tests that were relying on the very specific setup

## The next steps
Since this is only a PoC the goal wasn't to make all changes beautiful. Only to make things work.
Will follow up with a series of improvements.

### CODE REVIEW

File: e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js [hunk 1]
```diff
@@ -218,10 +218,8 @@ describe("scenarios > embedding > full app", () => {
           url: "/question/" + ORDERS_QUESTION_ID,
           qs: { side_nav: false, logo: false, breadcrumbs: false },
         });
-        cy.findAllByRole("banner")
-          .should("have.length", 1)
-          .findByText("Connect to your database to get the most from Metabase.")
-          .should("exist");
+        cy.findByDisplayValue("Orders");
```
Comment (alxnddr) [inline e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js:221]: Why do we need to change the test logic here and stop testing the db connection banner here?
Also, it seems that these specs should be EE with token

**Review by alxnddr — APPROVED** [review]
Looks awesome! 🙏🏻
I left one comment about a spec that has changed logic

File: e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js:221]: The test wasn't really about the top-most banner CTA "Connect to database...". I think @WiNloSt  added that to the test simply because we were always running with the premium token before and that's what you'd see with the token active.

The test simply makes sure that the main header (that contains the logo, breadcrumbs, etc.) doesn't show up if the right query parameters are set.

> Also, it seems that these specs should be EE with token

That is most likely true.
I haven't had the time to check every single feature's requirement and I plan to do that immediately after this overhaul. This spec will have to be re-written anyaway.

tl;dr
Instead of trying to do it all at once, the goal was to make tests in the current form work with the new logic first, and only then fix the correct behavior according to the feature flags.

Comment (WiNloSt) [reply e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js:221]: @nemanjaglumac 
> I think @WiNloSt added that to the test simply because we were always running with the premium token before and that's what you'd see with the token active.
>
> The test merely makes sure that the main header (that contains the logo, breadcrumbs, etc.) doesn't show up if the right query parameters are set.

That's very accurate. The point was to test there's no header except the DB prompt banner. For context, originally the test only tested that there was no banner at all using `findByRole("banner", ...)`. It was changed after when the database prompt banner was introduced since the database prompt banner will be found by `findByRole("banner", ...)` too.
